CoEpi/ble/Peripheral.swift

```diff
@@ -107,34 +117,26 @@ extension PeripheralImpl: CBPeripheralManagerDelegate {
 
     func peripheralManager(_ peripheral: CBPeripheralManager, didReceiveRead request: CBATTRequest) {
         os_log("Peripheral manager did receive read request: %@", log: blePeripheralLog, request.description)
+
+        let respondClosure: (Data) -> () = { [weak self] (responseData) in
+            // do we need to capture request & peripheral?
+            let theirData: Data? = request.value
+            request.value = responseData
+            peripheral.respond(to: request, withResult: .success)
+
+            let contact: Contact2 = .init(theirData: theirData, ourData: responseData)
+            self?.addNewContactEvent(contact: contact)
+        }
         
-        let currentCENKey = CENKey.generateAndStoreCENKey()
-        let CENData: Data = CEN.generateCENData(CENKey: currentCENKey.cenKey!)
-        //*** Scenario 1: https://docs.google.com/document/d/1f65V3PI214-uYfZLUZtm55kdVwoazIMqGJrxcYNI4eg/edit#
-        // iOS - Central + iOS - Peripheral -- so commenting out addNewContact
-        //addNewContactEvent(with: identifier)
-        request.value = CENData
-        peripheral.respond(to: request, withResult: .success)
-        
-        addNewContactEvent(with: currentCENKey.cenKey!)
+        internalDelegate.onReceivedRequest(request: request.value, respondClosure: respondClosure)
```

Comment:
Does this work if `onReceivedRequest` is an async/long running operation? It may depend only on the BLE timeout?

Reply:
It depends, I mean saving to DB is an I/O operation ("long running"), would we want to `peripheralContactSent` to emit an event before persistence finished? Or if persistence errored out? I actually don't know what the desired behavior is,.

Reply:
The actual question is for what are we persisting the generated CEN?